### PR TITLE
[SYCL][ESIMD] Add test for simd constructor from an array

### PR DESCRIPTION
**Pull request by vasilytric**
This test uses different data types, dimensionality and different `simd` constructor invocation contexts.
The test does the following actions:
- construct fixed-size array that filled with some values
- use `std::move()` to provide it to `simd` constructor
- bitwise compare expected and retrieved values

### CODE REVIEW

File: SYCL/ESIMD/api/functional/ctors/ctor_array.cpp [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+// context.
+struct var_decl {
+  static std::string get_description() { return "variable declaration"; }
+
+  template <typename DataT, int NumElems>
+  static void call_simd_ctor(DataT(&&ref_data)[NumElems], DataT *const out) {
+    static_assert(
+        type_traits::is_nonconst_rvalue_reference_v<decltype(ref_data)>,
+        "Provided input data is not nonconst rvalue reference");
+    simd<DataT, NumElems> simd_by_var_decl(std::move(ref_data));
```
Comment (yuriykoch) [inline SYCL/ESIMD/api/functional/ctors/ctor_array.cpp]:
```suggestion
    const simd<DataT, NumElems> simd_by_var_decl(std::move(ref_data));
```

File: SYCL/ESIMD/api/functional/ctors/ctor_array.cpp [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+  bool passed = true;
+
+  const auto types = get_tested_types<tested_types::all>();
+  const auto dims = get_all_dimensions();
+
+  passed &= for_all_types_and_dims<run_test, initializer>(types, dims, queue);
+  passed &= for_all_types_and_dims<run_test, var_decl>(types, dims, queue);
+  passed &= for_all_types_and_dims<run_test, rval_in_expr>(types, dims, queue);
+  passed &= for_all_types_and_dims<run_test, const_ref>(types, dims, queue);
```
Comment (yuriykoch) [inline SYCL/ESIMD/api/functional/ctors/ctor_array.cpp]:
```suggestion
  // Run for all combinations possible
  passed &= for_all_types_and_dims<run_test, initializer>(types, dims, queue);
  passed &= for_all_types_and_dims<run_test, var_decl>(types, dims, queue);
  passed &= for_all_types_and_dims<run_test, rval_in_expr>(types, dims, queue);
  passed &= for_all_types_and_dims<run_test, const_ref>(types, dims, queue);
```
Let's make the cases of all combinations used explicit

Comment (vasilytric) [reply SYCL/ESIMD/api/functional/ctors/ctor_array.cpp]: I've done it in this commit: 1c8437a80d85a14ddc0ec72719271031557b3dda

File: SYCL/ESIMD/api/functional/ctors/ctor_array.cpp (hunk 1 above)
Comment (vasilytric) [reply SYCL/ESIMD/api/functional/ctors/ctor_array.cpp]: I've done it in this commit: 1c8437a80d85a14ddc0ec72719271031557b3dda

**Comment by vasilytric** [comment]
@v-klochkov Vyacheslav, could you review this PR?

**Comment by vasilytric** [comment]
@v-klochkov I can't enable runtime verification due to `error: unsupported type for load/store` error retrieved with different data types. could you review this test with enabled test building option?

**Comment by v-klochkov** [comment]
> unsupported type for load/store

What are those "unsupported" types?

**Comment by vasilytric** [comment]
> What are those "unsupported" types?

@v-klochkov The test failed with simd vector length = 1. I've added macros for enabling tests for simd<DataT, 1> and now running test only with 8 and 16 simd vector length.

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**